console/src/components/Services/Data/Schema/ConnectDatabase.tsx

```diff
@@ -0,0 +1,769 @@
+
+    if (!host || !port || !username || !database) {
+      const errorMessage = getErrorMessageFromMissingFields(
+        host,
+        port,
+        username,
+        database
+      );
+      dispatch(
+        showErrorNotification('Some required fields are missing', errorMessage)
+      );
+      return;
+    }
+
+    const connectionURL = makeConnectionStringFromConnectionParams(
+      connectDBInputState.dbType,
+      host,
+      port,
+      username,
+      database,
+      password
+    );
+
+    dispatch(
+      addDataSource(
+        {
+          driver: connectDBInputState.dbType,
+          payload: {
+            name: connectDBInputState.displayName.trim(),
+            dbUrl: connectionURL,
+            connection_pool_settings: connectDBInputState.connectionSettings,
+          },
+        },
+        () => resetState()
+      )
+    );
+  };
+
+  return (
+    <RightContainer>
+      <Helmet
+        title={
+          isEditState ? 'Edit Database - Hasura' : 'Connect Database - Hasura'
+        }
+      />
+      <div className={`container-fluid ${styles.manage_dbs_page}`}>
+        <BreadCrumb breadCrumbs={crumbs} />
+        <div className={styles.padd_top}>
+          <div className={`${styles.display_flex} manage-db-header`}>
+            <h2 className={`${styles.headerText} ${styles.display_inline}`}>
+              Connect Database
+            </h2>
+          </div>
+        </div>
+        <hr />
+        <div className={styles.connect_db_content}>
+          <h4
+            className={`${styles.remove_pad_bottom} ${styles.connect_db_header}`}
+          >
+            Connect Database Via
+          </h4>
+          <div
+            className={styles.connect_db_radios}
+            onChange={onChangeConnectionType}
+          >
+            {connectionRadios.map(
+              (radioBtn: {
+                value: string;
+                title: string;
+                disableOnEdit: boolean;
+              }) => (
+                <label className={styles.connect_db_radio_label}>
+                  <input
+                    type="radio"
+                    value={radioBtn.value}
+                    name={connectionRadioName}
+                    checked={connectionType === radioBtn.value}
+                    disabled={
+                      isEditState === radioBtn.disableOnEdit && isEditState
+                    }
+                  />
+                  {radioBtn.title}
+                </label>
+              )
+            )}
+          </div>
+          <div className={styles.connect_form_layout}>
+            <label className={styles.connect_db_input_label}>
+              Database Name
+            </label>
+            <input
+              key="connect-display-name"
+              type="text"
+              onChange={e =>
+                connectDBDispatch({
+                  type: UPDATE_DISPLAY_NAME,
+                  data: e.target.value,
+                })
```

Comment:
Try changing this to the _classic_ way of handling multiple inputs: using name and the subsequent value.